src/nrnmpi/nrnmpi.cpp

```diff
@@ -145,6 +145,7 @@ for (i=0; i < *pargc; ++i) {
     nrnmpi_myid = nrnmpi_myid_bbs = nrnmpi_myid_world;
     nrnmpi_spike_initialize();
     nrnmpi_use = 1;
+    nrnmpi_subworld_change_cnt = 0;  // increment from within void nrnmpi_subworld_size(int n)
```

Comment:
Could this be just a boolean variable?

Reply:
During our discussion on Slack, we thought that having a counter would allow the MPI_Comm_split to be performed only when `subworlds` is invoked. If subworld_change_cnt is a boolean, then theoretically the user could call subworlds several times with different size arguments, but the split will be performed only the first time. And calling MPI_Comm_split each time psolve is called seemed excessive, so a subworlds invocation counter seemed like a reasonable compromise.

Reply:
I agree that a count variable is useful to avoid unnecessary calls to MPI_Comm_split

Reply:
Ah okay, I see now. Thanks for the clarification!